stackstorm-nexus3/actions/lib/base.py

```diff
@@ -0,0 +1,178 @@
+import sys
+from exception import *
+from nexuscli.repository import Repository
+from nexuscli.nexus_client import NexusClient
+from nexuscli.exception import *
+
+from st2common.runners.base_action import Action
+
+__all__ = [
+    'BaseAction'
+]
+
+REQUIRED_FIELDS = ['url', 'user', 'password']
+
+
+class BaseAction(Action):
+
+    def __init__(self, config):
+        super(BaseAction, self).__init__(config)
+        self.nexus_profiles = self.config.get('profiles', {})
+        self.nexus_client = None
+        self.dial_config = None
+
+    def validate_config(self, c_profile):
+        """ Validate connection configuration
+
+        """
+        validation_errors = []
+        is_valid = True
+        for field in REQUIRED_FIELDS:
+            if c_profile.get(field, None) == None:
+                validation_errors.push("%s if missing" % field)
+                is_valid = False
+
+        return (is_valid, validation_errors)
+
+    def get_connection_defaults(self):
+        """ default properties
+
+        """
+        return {
+            "url": self.config.get('url', None),
```

Comment:
Using `get()` already returns `None` if the key isn't found. This just makes it explicit.